### ISSUE TITLE
[Question] Reproduce about Poisson-Gauss

### ISSUE THREAD

**Issue by LilaKen** [camlab-ethz/GAOT#1]
Dear Shizheng Wen,

I hope this message finds you well. First of all, thank you very much for your excellent work on GAOT—I truly appreciate the methodological elegance and strong empirical results presented in your paper.

I recently attempted to reproduce the time-independent benchmarks using the official open-source codebase. Overall, the pipeline was clear and well-documented. However, I noticed a noticeable discrepancy in the Poisson-Gauss case: my reproduced median relative L1 error is approximately 2.03%, whereas the paper reports 0.83%. Given the magnitude of this gap, I’d like to kindly ask whether I might have overlooked any important configuration or training detail.

For reference, on the Elasticity task, I obtained a median relative L1 error of 1.53%, which is fairly close to the reported 1.34% and likely within reasonable variance.

The command I used (for Poisson-Gauss) was:

```bash
python main.py --config config/examples/time_indep/poisson_gauss.json
```

and analogously for elasticity.json, with no modifications to the default settings.

Could you please clarify whether the final results in the paper relied on additional details not included in the public config files? For example: specific random seeds, extended training epochs, learning rate scheduling, data normalization strategies, or other subtle training practices?

Any guidance would be greatly appreciated. Thank you again for your valuable contribution to the field!

Best regards,
LilaKen

**Comment by LilaKen** [comment]
I might know where the problem lies. The original paper "For time-independent PDEs we train for 1000 epochs unless otherwise noted. "should be because the epochs in poisson_gauss.json in the repository are 100. I'm very sorry. I'll try to increase the epochs again.

**Comment by LilaKen** [comment]
I retrained according to the 1000epochs situation in the paper. The result of Poisson-Gauss is 1.30, which is still a bit different from the original 0.83. Could you please tell me if there is something wrong with my Settings.

**Comment by LilaKen** [comment]
> I retrained according to the 1000epochs situation in the paper. The result of Poisson-Gauss is 1.30, which is still a bit different from the original 0.83. Could you please tell me if there is something wrong with my Settings.

| Dataset          | GAOT复现 | GAOT  | RIGNO-18 | Transolver | GNOT  | UPT   | GINO  |
|------------------|----------|-------|----------|------------|-------|-------|-------|
| **Time-Independent** |          |       |          |            |       |       |       |
| Poisson-C-Sines  | 15.42    | 3.10  | 6.83     | 77.30      | 100.00| 100.00| 20.00 |
| Poisson-Gauss    | **1.30**     | 0.83  | 2.26     | 2.02       | 88.90 | 48.40 | 7.57  |
| Elasticity       | **1.52**     | 1.34  | 4.31     | 4.92       | 10.40 | 12.60 | 4.38  |
| NACA0012         | 25.68    | 6.81  | 5.30     | 8.69       | 6.89  | 16.10 | 9.01  |
| NACA2412         | 26.22    | 6.66  | 6.72     | 8.51       | 8.82  | 17.90 | 9.39  |
| RAE2822          | 26.56    | 6.61  | 5.06     | 4.82       | 7.15  | 16.10 | 8.61  |
| Bluff-Body       | —        | 2.25  | 5.76     | 1.78       | 44.20 | 5.81  | 3.49  |
| **Time-Dependent**   |          |       |          |            |       |       |       |
| NS-Gauss         | **2.41**     | 2.91  | 2.29     | 41.10      | 38.40 | 92.50 | 13.10 |
| NS-PwC           | **1.36**     | 1.50  | 1.58     | 26.00      | 56.70 | 100.00| 5.85  |
| NS-SL            | 4.07     | 1.21  | 1.28     | 13.70      | 22.60 | 51.50 | 4.48  |
| NS-SVS           | **0.76**     | 0.46  | 0.56     | 9.75       | 26.00 | 4.20  | 1.19  |
| CE-Gauss         | 39.77    | 6.40  | 6.90     | 42.10      | 30.80 | 64.20 | 25.10 |
| CE-RP            | 11.43    | 5.97  | 3.98     | 18.40      | 27.70 | 26.80 | 12.30 |
| Wave-Layer       | 10.96    | 5.78  | 6.77     | 11.10      | 28.30 | 19.60 | 19.20 |
| Wave-C-Sines     | **4.79**     | 4.65  | 5.35     | 13.10      | 5.52  | 12.70 | 5.82  |

**Comment by Shizheng-Wen** [comment]
Hi Peijian,

Thanks for your attention to our work. 

Actually, we updated several versions of the GAOT repos and simplified the model for users to use. The performance is always stable and robust. So it is not exactly the same setup as the ones in our original paper. Actually, performance on some datasets should be even better, not even worse. For the NACA dataset, there is definitely a wrong setup for your experiments. I am traveling now and will be back on December 17. Then I will provide the detailed setup for these experiments.

Best,
Shizheng

**Comment by LilaKen** [comment]
Hi Shizheng,

Thank you so much for your prompt and thoughtful response!

Just to clarify—aside from the provided config files, for the NACA experiments, I simply copied the config file from the corresponding directory and only modified the dataset section as follows:

```json
"dataset": {
    "name": "NACA0012",
    "metaname": "compressible_flow/naca0012"
}
```
I completely agree that the issue likely lies in my experimental setup, but I haven’t been able to pinpoint the exact cause yet. I’d be really grateful to learn from you once you’re back! Your team’s work has been incredibly inspiring to me, and I truly admire the thoughtfulness and robustness built into GAOT.


Wishing you a wonderful and relaxing trip—hope you’re having a great time! Safe travels, and looking forward to your insights after December 17.

Best regards,
Peijian

**Comment by LilaKen** [comment]
Hi Shizheng,

Hope you’ve had a wonderful and relaxing time on your recent trip! Traveling can be such a great way to recharge and gain fresh perspectives—whether you were exploring new places, enjoying some quiet time, or catching up with friends and family.

I know you’ve been busy, so no rush at all—but just wanted to gently follow up and let you know I’m really looking forward to hearing back from you whenever you’re settled back in and have a moment. There’s no urgency, but I’d greatly appreciate your thoughts whenever you get the chance.

Wishing you a smooth transition back to work and daily routines!

Best regards,
Peijian

**Comment by Shizheng-Wen** [comment]
Hi Peijian,

Sorry for the late reply :-) I have been busy with the TA assignments for the exam in the last two days. But I want to sit down and try to figure out everything with you. I don't want you to wait too long  :-)

I am really happy that you are trying to replicate the results! This can also help me fix the code of the repo in the next version. As I said before, this is a simplified version of the original GAOT (as you can see, the number of trainable parameters for the model is totally different from the paper). I didn't check all the experiments under this setup, because I just want to provide an easy version for the user to test. But no worry, I kept all the ckpt and config in my internal version of the original version.

Let me discuss a potential reason behind it. 
For NACA0012, NACA2412, and RAE2822, did you use the correct number of training samples? In the original paper, we use a train/validation/test split of 5000/128/256. Then, for these datasets with variable coordinates, only use geometric embedding in the encoder, do not use it in the decoder, the performance will be further improved a lot (also mentioned in the paper, but in this code base, I notice that I didn't provide this option, I will update it in the next version)
For the Poisson-C-Sines dataset, this dataset is kind of tricky; the model needs to use the multi-scale features, otherwise, it is even harder to converge. Please note that, just for simplicity, I only use one scale in the provided config files. Try to use node embedding and multi-scale radius. 

For a time-dependent dataset, the performance for some of them is too large. But there are several reasons:
1. We only calculate the error of the final time step, and implement three different stepper modes: [link](https://github.com/camlab-ethz/GAOT/blob/549c5a5f7113e23ba5e91469f2f8bbb1567fae46/src/trainer/sequential_trainer.py#L377C9-L377C27). And choose the best one of each model to represent its error. The following figure is the mode that we select for each model. E.g., for the wave equation dataset, the best mode is always direct prediction.

<img width="934" height="205" alt="Image" src="https://github.com/user-attachments/assets/91455ce5-b5eb-4e87-8874-04d5c74dc15b" />

2. Please use node embedding for a time-dependent dataset; you will find that the performance can be improved :-)

3. Using more layers in the processor for CE-RP and CE-Gauss can improve the performance again. Try 10 layers for these two datasets and see what will happen.

Due to the maintenance of the ETH Euler cluster, I cannot test by myself immediately. But as I said before, I will fix this problem and update it in the next version of the repo. Thanks for your careful inspection, and I really appreciate it. Btw, I didn't do any parameter tuning of GAOT, because in my experiment, the performance is always pretty stable. If you find something even better or worse, please feel free to tell me. Always feel free to send me a message at any time! 

Best,
Shizheng

**Comment by LilaKen** [comment]
Hi Shizheng,

Thank you so much for your detailed and thoughtful reply!   I really appreciate you taking the time to explain all these points while you’re busy handling TA assignments for the exam.

Please don’t feel rushed on my account—I’m in no hurry at all.   On the contrary, I feel truly honored to be working with your code and hope I can contribute something useful to your repository, even if just a little.

I’ll go ahead and try out your suggestions—like using the correct train/val/test split, enabling node embedding, and experimenting with multi-scale features—and see how things improve.

Thanks again for your kindness and support amidst your busy TA duties.   Wishing you smooth grading and a bit of rest soon!

Best regards,
Peijian

**Comment by LilaKen** [comment]
Hi Shizheng,

Thank you so much for your detailed and helpful response! I really appreciate you taking the time to clarify these points despite your busy schedule.

Following your suggestions, I re-ran the experiments with the corrected data split (5000/128/256) and enabled geometric embedding **only in the encoder** (not in the decoder). This indeed led to a significant improvement — for example, the NACA0012 error dropped from ~25 to ~12. However, even after training for 2000 epochs, the results still fall short of those reported in the paper (e.g., NACA0012: 12.07 vs. paper-reported ~6.8). Below are my current results for reference:

| Dataset       | Ep 2000 (Dec F) | GAOT (Paper) |
|---------------|------------------|---------------|
| NACA0012      | 12.07            | 6.81          |
| NACA2412      | 11.80            | 6.66          |
| RAE2822       | 13.22            | 6.61          |
| Wave-Layer (DR) | 10.96 | 5.78          |

For the **Poisson-C-Sines** dataset, using node embedding + multi-scale radius with 2000 epochs gave me **5.27** (vs. 10.20 at 1000 epochs, and 15.42 without multi-scale). While this is a big step forward, it’s still above the paper’s reported **3.10**.

Given these gaps, I suspect the discrepancy may stem from differences in the model architecture or training configuration in the simplified public repo (e.g., number of parameters, processor depth, embedding dimensions, or even loss weighting). Since you mentioned you have the full internal version with all checkpoints and configs, I’d be very grateful if, when convenient, you could share any additional details (e.g., exact processor layer count, hidden dimensions, or whether gradient clipping/learning rate scheduling was used) that might help bridge this gap.

No rush at all — I completely understand the cluster maintenance and your other commitments! Just wanted to share my findings and thank you again for your transparency and support. Looking forward to the next version of the repo!

Best regards,  
Peijian
